File: src/solution_java/medium/_71SimplifyPath.java
package solution_java.medium;


import java.util.LinkedList;
import java.util.List;

public class _71SimplifyPath {
    public class Solution {
        public String simplifyPath(String path) {
            if (path == null || path.isEmpty()) {
                return "/";
            }
            String[] array = path.split("/");
            List<String> list = new LinkedList<>();
            for (int i = 0; i < array.length; i++) {
                String item = array[i];
                if (item.equals("..")) {
                    if (!list.isEmpty()) {
                        list.remove(list.size() - 1);
                    }
                } else if (!item.equals(".") && !item.isEmpty()) {
                    list.add(item);
                }
            }
            StringBuilder sb = new StringBuilder("/");
            int size = list.size();
            for (int i = 0; i < size; i++) {
                String item = list.get(0);
                sb.append(item);
                list.remove(0);
                if (!list.isEmpty()) {
                    sb.append("/");
                }
            }
            return sb.toString();
        }
    }
}

File: src/medium/_71SimplifyPath.kt
package medium

import java.util.*

class _71SimplifyPath {
    class Solution {
        fun simplifyPath(path: String): String {
            if (path.isEmpty()) {
                return "/"
            }
            val array = path.split("/")
            val list = LinkedList<String>()
            for (i in array.indices) {
                val item = array[i]
                if (item.isEmpty() || item == ".") {
                    continue
                } else if (item == "..") {
                    if (list.isNotEmpty()) {
                        list.pollLast()
                    }
                } else {
                    list.offer(item)
                }
            }
            val sb = StringBuffer("/")
            for (i in list.indices) {
                val item = list.pollFirst()
                sb.append(item)
                if (list.isNotEmpty()) {
                    sb.append("/")
                }
            }
            return sb.toString()
        }
    }
}